vyper/optimizer.py

```diff
@@ -11,7 +11,7 @@ def get_int_at(args, pos, signed=False):
         o = LOADED_LIMIT_MAP[args[pos].args[0].value]
     else:
         return None
-    if signed:
+    if signed or o < 0:
```

Comment:
This is the hackiest line in this PR. I had to do this to prevent `-1` getting turned into `2**256 - 1 = 115792089237316195423570985008687907853269984665640564039457584007913129639935`, which would then get added to `999999999` and overflow when optimising `ceil(-0.0000000001)`.

The other solution I considered was making the additions (and other operations) carried out by the optimiser modulo `2**256` like their equivalent EVM instructions, but I figured this was sub-optimal as it silently hides the overflow, rather than preventing it.